tests/test_psi_phi_array.py

```diff
@@ -174,6 +175,35 @@ def test_fill_psi_phi_array(self):
                 self.assertFalse(arr.gpu_array_allocated)
                 self.assertFalse(arr.gpu_time_array_allocated)
 
+    def test_fill_invalid(self):
+        arr = PsiPhiArray()
+
+        # Fails with NaN
+        self.psi_1.set_pixel(4, 3, math.nan)
+        self.assertRaises(
+            RuntimeError,
+            fill_psi_phi_array,
+            arr,
+            4,
+            [self.psi_1, self.psi_2],
+            [self.phi_1, self.phi_2],
+            self.zeroed_times,
+            False,
+        )
+
+        # Fails with inf
+        self.psi_1.set_pixel(4, 3, math.inf)
```

Comment:
It would also be good to test for invalid phi values (might have caught the above typo). A coverage check might have also revealed that the line that raises an error was not covered by tests?